server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -179,74 +186,78 @@ public void addSegment(DataSegment segment, DataSegmentChangeCallback callback)
     }
   }
 
-  public void addSegments(Iterable<DataSegment> segments, final DataSegmentChangeCallback callback)
+  private void addSegments(Collection<DataSegment> segments, final DataSegmentChangeCallback callback)
```

Comment:
I'm not sure I understand.  As it is, failure to load one segment doesn't kill the whole startup and we definitely want this method to not return until all of the segment have been added.